legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java

```diff
@@ -130,14 +130,24 @@ public static Condition getPartitionColumnsMatchCondition(Dataset mainDataSet, D
         return getColumnsMatchCondition(mainDataSet, stagingDataSet, partitionColumns);
     }
 
-    public static Condition getPartitionColumnsDoNotMatchCondition(Dataset mainDataSet, Dataset stagingDataSet, String[] partitionColumns)
+    public static Condition getPartitionColumnValueMatchInCondition(Dataset dataSet, Map<String, Set<String>> partitionFilter)
     {
-        return getColumnsDoNotMatchCondition(mainDataSet, stagingDataSet, partitionColumns);
+        return getColumnValueMatchInCondition(dataSet, partitionFilter);
     }
 
-    public static Condition getPartitionColumnValueMatchInCondition(Dataset dataSet, Map<String, Set<String>> partitionFilter)
+    // (key1 = val11 AND key2 = val21) OR (key1 = val12 AND key2 = val22) OR ...
+    public static Condition getPartitionSpecMatchCondition(Dataset dataSet, List<Map<String, String>> partitionSpecList)
     {
-        return getColumnValueMatchInCondition(dataSet, partitionFilter);
+        return Or.of(partitionSpecList.stream()
+                 .map(partitionSpec -> And.of(
+                         partitionSpec.entrySet().stream()
+                                 .map(columnValuePair ->
+                                         Equals.of(
+                                                 FieldValue.builder().datasetRef(dataSet.datasetReference()).fieldName(columnValuePair.getKey()).build(),
+                                                 StringValue.of(columnValuePair.getValue()))
```

Comment:
May be a good chance to fix this one: here it may not always be a StringValue, it could be an ObjectValue when the field is a Number (so that we do not print the quotes around the value); similarly we can fix this one for partitionValuesByField case too

Reply:
In that case the interface needs to be a List<Map<String, Object>>. 
Don't think we can fix for partitionValuesByField as the type of value is String there
Map<String, Set<String>> partitionValuesByField();